tests/console/commands/self/test_add_plugins.py

```diff
@@ -128,14 +130,11 @@ def test_add_with_git_constraint_with_extras(
 Writing lock file
 """
 
-    assert_plugin_add_result(
-        tester,
-        expected,
-        {
-            "git": "https://github.com/demo/poetry-plugin.git",
-            "extras": ["foo"],
-        },
-    )
+    constraint: dict[str, Any] = {
```

Comment:
couldn't this `Any` be made into `str | list[str]`?

Reply:
that leads to a small rabbit-hole where it disagrees with `assert_plugin_add_result`.

I have a few more type annotation changes coming, I'd suggest merge this as it is now and I pinky-promise to do more with it in the next (final?) round

Reply:
actually one last push and we're done...